src/cleaners/asCodec.js

```diff
@@ -1,18 +1,35 @@
-let uncleaning = 0
-
 export function asCodec(cleaner, uncleaner) {
   return function asCodec(raw) {
-    return uncleaning > 0 ? uncleaner(raw) : cleaner(raw)
+    return global[uncleaning] > 0 ? uncleaner(raw) : cleaner(raw)
   }
 }
 
 export function uncleaner(cleaner) {
   return function uncleaner(raw) {
+    // We need to set a flag to indicate that we are un-cleaning.
+    // The flag needs to be in a global location,
+    // since multiple cleaner implementations should agree
+    // which mode we are in.
+    if (global[uncleaning] == null) {
+      Object.defineProperty(global, uncleaning, {
+        value: 0,
+        writable: true
+      })
+    }
     try {
-      ++uncleaning
+      ++global[uncleaning]
       return cleaner(raw)
     } finally {
-      --uncleaning
+      --global[uncleaning]
     }
   }
 }
+
+// If we cannot find the standardized "globalThis" object,
+// fall back on using the "JSON" constructor, which is also well-known.
+const global =
+  typeof globalThis === 'object' && globalThis != null ? globalThis : JSON
+
+/* istanbul ignore next */
+const uncleaning =
+  typeof Symbol === 'function' ? Symbol.for('uncleaning') : 'uncleaning'
```

Comment:
Shouldn't the fallback identifier be a bit more obscure. It's unlikely that we'll have a collision, but it would be even more unlikely if it were something like `CLEANER_UNCLEANING_FLAG` or something crazy like that. Not super important of a change.